cg/meta/workflow/analysis.py

```diff
@@ -495,10 +503,53 @@ def is_case_ready_for_analysis(self, case_id: str) -> bool:
             return False
         return True
 
+    def does_any_file_need_to_be_retrieved(self, case_id: str) -> bool:
+        """Checks whether we need to retrieve files from an external data location."""
+        if self.get_archive_location_for_case(case_id) == ArchiveLocations.PDC:
+            if self._is_flow_cell_check_applicable(
+                case_id
+            ) and not self.status_db.are_all_flow_cells_on_disk(case_id):
+                LOG.warning(f"Case {case_id} is not ready - all flow cells not present on disk.")
+                return True
+        else:
+            if not self.are_all_spring_files_present(case_id):
+                LOG.warning(f"Case {case_id} is not ready - some files are archived.")
+                return True
+        return False
```

Comment:
I think this can be simplified a bit. I think we should check things in the following order:

1. `if self._is_flow_cell_check_applicable(case_id) and not self.status_db.are_all_flow_cells_on_disk(case_id)`
```suggestion
        if self._is_flow_cell_check_applicable(case_id) and not self.status_db.are_all_flow_cells_on_disk(case_id):
                LOG.warning(f"Case {case_id} is not ready - all flow cells not present on disk.")
                return True
        else:
            if not self.are_all_spring_files_present(case_id):
                LOG.warning(f"Case {case_id} is not ready - some files are archived.")
                return True
        return False
```

Since we in the past have sent all flow cells to PDC we cannot look at `ArchiveLocations` at this point

Reply:
Great catch! I think I was writing this with some utopian end game in mind 😅 Will this still be the case after we have run the initial sync where we retrieve clinical data from PDC and archive via DDN? 

Also - do we not need to modify `are_all_flow_cells_on_disk`? Or do we intend to not modify the flow cells status when archiving the files?

Reply:
Yeah, It will be the case until we've re-fetched every flow cell we've sent to PDC... I don't think we should modify the flow cell status. when we've fetched all flow cells (in the far future) we can remove the column

Reply:
Which column are we talking about now? Will we not need to support PDC for research customer on an indefinite time scale?

Reply:
The status column for flow cells. But that is something we can take in the far future